npm/test/basics.js

```diff
@@ -879,3 +880,251 @@ test("debug service compiling multiple sources - web worker", async () => {
+});
+
+test("CreateTimeTicks: months", () => {
+  runAndAssertTimeTicks(week - 1, month, ["week", "month"]);
+  runAndAssertTimeTicks(month - 1, 2 * month, ["month"]);
+  runAndAssertTimeTicks(month, 11 * month, ["month"]);
+  runAndAssertTimeTicks(month, 12 * month, ["month"]);
+  runAndAssertTimeTicks(month, 12 * month + 5 * day, ["month", "year"]);
+  runAndAssertTimeTicks(month + 1, 12 * month, ["10 months"]);
+  // due to precision issues month + 1 == month
+  runAndAssertTimeTicks(month + hour, 10 * month - hour, [
+    "2 months",
+    "3 months",
+    "4 months",
+    "5 months",
+    "6 months",
+    "7 months",
+    "8 months",
+    "9 months",
+  ]);
+  runAndAssertTimeTicks(week, 111 * month, ["week", "month", "year"]);
+  runAndAssertTimeTicks(month + 1, 111 * month, ["year"]);
+});
+
+test("CreateTimeTicks: years", () => {
+  runAndAssertTimeTicks(month - 1, year, ["month", "year"]);
+  runAndAssertTimeTicks(year - month, 2 * year, ["year"]);
+  // due to precision issues year + 1 == year and decade - 1 == decade
+  runAndAssertTimeTicks(year + day, decade - day, [
+    "2 years",
+    "3 years",
+    "4 years",
+    "5 years",
+    "6 years",
+    "7 years",
+    "8 years",
+    "9 years",
+  ]);
+
+  runAndAssertTimeTicks(month, 111 * year, [
+    "month",
+    "year",
+    "decade",
+    "century",
+  ]);
+});
+
+test("CreateTimeTicks: decades", () => {
+  // due to precision issues year + 1 == year
+  runAndAssertTimeTicks(year + day, 21 * year, ["decade"]);
+  runAndAssertTimeTicks(year, decade, ["year", "decade"]);
+  runAndAssertTimeTicks(decade - year, 2 * decade, ["decade"]);
+  runAndAssertTimeTicks(year, 111 * decade, ["year", "decade", "century"]);
+  // due to precision issues decade + 1 == decade
+  runAndAssertTimeTicks(decade + month, 111 * decade, ["century"]);
+});
+
+test("CreateTimeTicks: centuries", () => {
+  runAndAssertTimeTicks(decade - 1, century, ["decade", "century"]);
+  runAndAssertTimeTicks(century - decade, 2 * century, ["century"]);
+  runAndAssertTimeTicks(century, 11 * century, ["century"]);
+  runAndAssertTimeTicks(century + 1, 21 * century, ["century"]);
+  runAndAssertTimeTicks(decade, 111 * century, ["decade", "century"]);
+  runAndAssertTimeTicks(century + 1, 111 * century, ["century"]);
+});
+
+test("CreateTimeTicks: above centuries", () => {
+  runAndAssertTimeTicks(century + 30 * year, 3 * century, [
+    "2 centuries",
+    "3 centuries",
+  ]);
+  runAndAssertTimeTicks(century + 30 * year, century + 55 * year, [
+    "13 decades",
+    "14 decades",
+    "15 decades",
+  ]);
+  runAndAssertTimeTicks(2 * century + 32 * year, 2 * century + 36 * year, [
+    "232 years",
+    "233 years",
+    "234 years",
+    "235 years",
+    "236 years",
+  ]);
+});
+
+const second = 1e9;
+const minute = 60 * second;
+const hour = 60 * minute;
+const day = 24 * hour;
+const week = 7 * day;
+const month = 30 * day;
+const year = 365 * day;
+const decade = 10 * year;
+const century = 10 * decade;
```

Comment:
For readability, can we have all these consts above all the code where they're used, rather than after.